avatars/avatars.js

```diff
@@ -1418,6 +1421,24 @@ class Avatar {
       }
     }
   }
+  lastVelocity = new THREE.Vector3();
+  setVelocity(timeDiffS, lastPosition, currentPosition, currentQuaternion) {
+    // Set the velocity, which will be considered by the animation controller
+    const positionDiff = localVector.copy(lastPosition)
+      .sub(currentPosition)
+      .divideScalar(timeDiffS)
+      .multiplyScalar(0.1);
+    localEuler.setFromQuaternion(currentQuaternion, 'YXZ');
+    localEuler.x = 0;
+    localEuler.z = 0;
+    localEuler.y += Math.PI;
+    localEuler2.set(-localEuler.x, -localEuler.y, -localEuler.z, localEuler.order);
```

Comment:
FYI you don't need a second euler for this. You are already mutating the existing one anyway.